lib/github.ml

```diff
@@ -323,41 +323,45 @@ let assert_tag_exists ~dry_run tag =
    point to the same ref. *)
 let push_tag ~dry_run ~yes ~dev_repo vcs tag =
   let remote_has_tag_uptodate () =
```

Comment:
How about we move this function to the toplevel of the module and also eventually extract other functions so its flow reads better? It was already quite convoluted and the triple level of nested pattern matching it has now didn't really make it any better.

Reply:
I tried refactoring it a bit. In general the way it determines a tag is annotated is kinda weird and I feel that functionality should eventually live somewhere else but I also think that that's a battle for another time.